clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -586,7 +586,7 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
             log.trace("Cancelled request with header {} due to the last request to node {} timed out",
                 requestHeader, response.destination());
             for (ProducerBatch batch : batches.values())
-                completeBatch(batch, new ProduceResponse.PartitionResponse(Errors.REQUEST_TIMED_OUT, String.format("Disconnected from node %s due to timeout", response.destination())),
+                completeBatch(batch, new ProduceResponse.PartitionResponse(Errors.NETWORK_EXCEPTION, String.format("Disconnected from node %s due to request timeout", response.destination())),
```

Comment:
Please add a comment clarifying that this cannot be REQUEST_TIMEOUT even though the message says so